src/App.js

```diff
@@ -16,8 +16,27 @@ class App extends Component {
         name: "Wookie",
       },
     ],
+    newDogName: "",
   };
 
+  handleChangeDogName = (event) => {
+    this.setState({ newDogName: event.target.value });
+  }
+
+  handleAdoptClick = () => {
+    let newDog = {
+      name: this.state.newDogName
+    }
+
+    this.setState((state) => {
+      if (this.state.newDogName !== "") {
```

Comment:
make sure to use `state` here instead of `this.state`. since you are inside this cb function